vividus-plugin-azure-storage-account/src/main/java/org/vividus/azure/storage/blob/BlobStorageSteps.java

```diff
@@ -231,6 +236,100 @@ public void findBlobs(StringComparisonRule rule, String blobNameToMatch, String
+     *                          <br>in container `global` of storage account `storage`
+     *                          <br> and save result to story variable `blobs`
+     *                          </code>
+     *                          <br>
+     *                          <br>where all filters are optional
+     *                          <ul>
+     *                           <li><code>blobNameFilterRule</code> The blob name comparison rule: "is equal to",
+     *                           "contains", "does not contain" or "matches".
+     *                           Should be specified along with <i>blobNameFilterValue</i>.</li>
+     *                           <li><code>blobNameFilterValue</code> The full or partial blob name to be matched.
+     *                           Should be specified along with <i>blobNameFilterRule</i>.</li>
+     *                           <li><code>blobNamePrefix</code> The directory(prefix) to list blobs underneath.</li>
+     *                           <li><code>resultsLimit</code> Maximum number of blob names to return.</li>
+     *                          </ul>
+     * @param containerName     The name of the container to point to.
+     * @param storageAccountKey The key to Storage Account endpoint.
+     * @param scopes            The set (comma separated list of scopes e.g.: STORY, NEXT_BATCHES) of the variable
+     *                          scopes.<br>
+     *                          <i>Available scopes:</i>
+     *                          <ul>
+     *                          <li><b>STEP</b> - the variable will be available only within the step,
+     *                          <li><b>SCENARIO</b> - the variable will be available only within the scenario,
+     *                          <li><b>STORY</b> - the variable will be available within the whole story,
+     *                          <li><b>NEXT_BATCHES</b> - the variable will be available starting from next batch
+     *                          </ul>
+     * @param variableName      The variable name to store the list of found blob names.
+     */
+    @When("I filter blobs by:$filters in container `$containerName` of storage account `$storageAccountKey`"
+            + " and save result to $scopes variable `$variableName`")
+    public void findBlobsByFilters(List<BlobFilter> filters, String containerName,
+                                      String storageAccountKey, Set<VariableScope> scopes, String variableName)
+    {
+        List<BlobFilter> checkedFilters = checkFilters(filters);
+        List<String> blobNames = new ArrayList<>();
+        BlobContainerClient blobContainerClient = createBlobContainerClient(containerName, storageAccountKey);
+        for (BlobFilter filter : checkedFilters)
+        {
+            blobContainerClient
+                    .listBlobsByHierarchy(filter.getBlobNamePrefix().orElse(StringUtils.EMPTY))
```

Comment:
why is an empty string used as default value? we shouldn't set this filter if no option is specified to avoid relying Azure SDK defaults